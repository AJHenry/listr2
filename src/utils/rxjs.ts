export { Observable } from 'rxjs'
export { Subject } from 'rxjs'
