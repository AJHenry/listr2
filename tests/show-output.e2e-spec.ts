import delay from 'delay'
import { mockProcessExit, mockProcessStderr, mockProcessStdout } from 'jest-mock-process'
import { Observable } from 'rxjs'

import { Listr } from '@root/index'

describe('show output from task', () => {
  let mockExit: jest.SpyInstance<never, [number?]>

  // eslint-disable-next-line @typescript-eslint/ban-types
  let mockStdout: jest.SpyInstance<boolean, [string, string?, Function?]>

  // eslint-disable-next-line @typescript-eslint/ban-types
  let mockStderr: jest.SpyInstance<boolean, [string, string?, Function?]>

  beforeEach(async () => {
    mockExit = mockProcessExit()
    mockStdout = mockProcessStdout()
    mockStderr = mockProcessStderr()
  })

  afterEach(async () => {
    mockExit.mockRestore()
    mockStdout.mockRestore()
    mockStderr.mockRestore()
    jest.clearAllMocks()
  })

  // test-e2e-rMG224TBrLk3ocYtKidc1D4AyZtEHm11
  it('should yield example output', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-rMG224TBrLk3ocYtKidc1D4AyZtEHm11-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-rMG224TBrLk3ocYtKidc1D4AyZtEHm11-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-rMG224TBrLk3ocYtKidc1D4AyZtEHm11-exit')
  })

  // test-e2e-oYHBlOYGg8juKRkaqigY617eyLbGMuDd
  it('should have persistent output', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { persistentOutput: true }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-oYHBlOYGg8juKRkaqigY617eyLbGMuDd-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-oYHBlOYGg8juKRkaqigY617eyLbGMuDd-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-oYHBlOYGg8juKRkaqigY617eyLbGMuDd-exit')
  })

  // test-e2e-767BkeBTfR1lrS2ANYYH7CLWPATxqyat
  it('should output to bottom bar', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { bottomBar: Infinity }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-767BkeBTfR1lrS2ANYYH7CLWPATxqyat-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-767BkeBTfR1lrS2ANYYH7CLWPATxqyat-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-767BkeBTfR1lrS2ANYYH7CLWPATxqyat-exit')
  })

  // test-e2e-JMCRBo4OtLm7JB3XbcYoRcCdQRiKfPdP
  it('should output to bottom bar from task with no title', async () => {
    await new Listr(
      [
        {
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { bottomBar: Infinity }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-JMCRBo4OtLm7JB3XbcYoRcCdQRiKfPdP-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-JMCRBo4OtLm7JB3XbcYoRcCdQRiKfPdP-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-JMCRBo4OtLm7JB3XbcYoRcCdQRiKfPdP-exit')
  })

  // test-e2e-iI52S3WPytorU9EZKPar2AiBrLAZTVut
  it('should output to bottom bar from task with no title persistently', async () => {
    await new Listr(
      [
        {
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { bottomBar: Infinity, persistentOutput: true }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-iI52S3WPytorU9EZKPar2AiBrLAZTVut-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-iI52S3WPytorU9EZKPar2AiBrLAZTVut-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-iI52S3WPytorU9EZKPar2AiBrLAZTVut-exit')
  })

  // test-e2e-HhZEM7noGNW4xpgxv4ZtXsPMroPWqrEA
  it('should output to bottom bar persistently', async () => {
    await new Listr(
      [
        {
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { bottomBar: Infinity, persistentOutput: true }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-HhZEM7noGNW4xpgxv4ZtXsPMroPWqrEA-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-HhZEM7noGNW4xpgxv4ZtXsPMroPWqrEA-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-HhZEM7noGNW4xpgxv4ZtXsPMroPWqrEA-exit')
  })

  // test-e2e-d4wXg4yGYak09qivTqgKFZaQJ3PvDZm5
  it('should limit output to bottom bar persistently', async () => {
    await new Listr(
      [
        {
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { bottomBar: 2 }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-d4wXg4yGYak09qivTqgKFZaQJ3PvDZm5-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-d4wXg4yGYak09qivTqgKFZaQJ3PvDZm5-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-d4wXg4yGYak09qivTqgKFZaQJ3PvDZm5-exit')
  })

  // test-e2e-NpGb4ry8b6hlK7VkJ1YcXcVibx0k5Sus
  it('should output to bottom bar 2 times at most and delete the prior tasks output when finished', async () => {
    await new Listr(
      [
        {
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { bottomBar: true, persistentOutput: false }
        },

        {
          task: async (ctx, task): Promise<void> => {
            task.output = 'I will push an output. [0]'
            await delay(5)

            task.output = 'I will push an output. [1]'
            await delay(5)

            task.output = 'I will push an output. [2]'
            await delay(5)
          },
          options: { bottomBar: true, persistentOutput: true }
        }
      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-NpGb4ry8b6hlK7VkJ1YcXcVibx0k5Sus-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-NpGb4ry8b6hlK7VkJ1YcXcVibx0k5Sus-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-NpGb4ry8b6hlK7VkJ1YcXcVibx0k5Sus-exit')
  })

  // test-e2e-SM8IHVdptzrFs7Qk2bseYbdCwtTf03QT
  it('should output to from an observable', async () => {
    await new Listr(
      [
        {
          title: 'Observable test.',
          task: (): Observable<string> =>
            new Observable((observer) => {
              observer.next('test')

              delay(5)
                .then(() => {
                  observer.next('changed')
                  return delay(5)
                })
                .then(() => {
                  observer.complete()
                })
            })
        }

      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-SM8IHVdptzrFs7Qk2bseYbdCwtTf03QT-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-SM8IHVdptzrFs7Qk2bseYbdCwtTf03QT-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-SM8IHVdptzrFs7Qk2bseYbdCwtTf03QT-exit')
  })

  // test-e2e-j7BqsosH97ffW1SQSdkADSm2HnSZQ9nn
  it('should indent long multiline output with persistent output', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: async (ctx, task): Promise<void> => {
            const start = 'This is a'
            const mid = 'long '
            const end = 'multi line output.'
            task.output = start + mid.repeat(40) + '\n'+ mid.repeat(40) + '\n'+ mid.repeat(40) + '\n'+ mid.repeat(40) + '\n'+ mid.repeat(40) + '\n' + end
            await delay(500)
          },
          options: { persistentOutput: true }
        }

      ],
      {
        concurrent: false, nonTTYRenderer: 'default', nonTTYRendererOptions: { lazy: true }
      }
    ).run()

    expect(mockStdout.mock.calls).toMatchSnapshot('test-e2e-j7BqsosH97ffW1SQSdkADSm2HnSZQ9nn-out')
    expect(mockStderr.mock.calls).toMatchSnapshot('test-e2e-j7BqsosH97ffW1SQSdkADSm2HnSZQ9nn-err')
    expect(mockExit.mock.calls).toMatchSnapshot('test-e2e-j7BqsosH97ffW1SQSdkADSm2HnSZQ9nn-exit')
  })

})
