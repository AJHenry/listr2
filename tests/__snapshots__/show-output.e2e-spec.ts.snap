// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show output from task should have persistent output: test-e2e-oYHBlOYGg8juKRkaqigY617eyLbGMuDd-err 1`] = `Array []`;

exports[`show output from task should have persistent output: test-e2e-oYHBlOYGg8juKRkaqigY617eyLbGMuDd-exit 1`] = `Array []`;

exports[`show output from task should have persistent output: test-e2e-oYHBlOYGg8juKRkaqigY617eyLbGMuDd-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
  ‚Ä∫ I will push an output. [0]  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
  ‚Ä∫ I will push an output. [1]  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
  ‚Ä∫ I will push an output. [2]  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
  ‚Ä∫ I will push an output. [2]  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
  ‚Ä∫ I will push an output. [2] 
",
  ],
]
`;

exports[`show output from task should indent long multiline output with persistent output: test-e2e-j7BqsosH97ffW1SQSdkADSm2HnSZQ9nn-err 1`] = `Array []`;

exports[`show output from task should indent long multiline output with persistent output: test-e2e-j7BqsosH97ffW1SQSdkADSm2HnSZQ9nn-exit 1`] = `Array []`;

exports[`show output from task should indent long multiline output with persistent output: test-e2e-j7BqsosH97ffW1SQSdkADSm2HnSZQ9nn-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
  ‚Ä∫ This is along long long long long long long long long long long long long l‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    multi line output.  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
  ‚Ä∫ This is along long long long long long long long long long long long long l‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    multi line output.  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
  ‚Ä∫ This is along long long long long long long long long long long long long l‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    long long long long long long long long long long long long long long long ‚Ä¶
    multi line output. 
",
  ],
]
`;

exports[`show output from task should limit output to bottom bar persistently: test-e2e-d4wXg4yGYak09qivTqgKFZaQJ3PvDZm5-err 1`] = `Array []`;

exports[`show output from task should limit output to bottom bar persistently: test-e2e-d4wXg4yGYak09qivTqgKFZaQJ3PvDZm5-exit 1`] = `Array []`;

exports[`show output from task should limit output to bottom bar persistently: test-e2e-d4wXg4yGYak09qivTqgKFZaQJ3PvDZm5-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[32m‚úî[39m This task will execute. 
",
  ],
]
`;

exports[`show output from task should output to bottom bar 2 times at most and delete the prior tasks output when finished: test-e2e-NpGb4ry8b6hlK7VkJ1YcXcVibx0k5Sus-err 1`] = `Array []`;

exports[`show output from task should output to bottom bar 2 times at most and delete the prior tasks output when finished: test-e2e-NpGb4ry8b6hlK7VkJ1YcXcVibx0k5Sus-exit 1`] = `Array []`;

exports[`show output from task should output to bottom bar 2 times at most and delete the prior tasks output when finished: test-e2e-NpGb4ry8b6hlK7VkJ1YcXcVibx0k5Sus-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[2m‚óº[22m This task will execute.
[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
[2m‚óº[22m This task will execute. 

‚Ä∫ I will push an output. [0] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
[2m‚óº[22m This task will execute. 

‚Ä∫ I will push an output. [1] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
[2m‚óº[22m This task will execute. 

‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
[33m‚ùØ[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [1] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
[32m‚úî[39m This task will execute. 

‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.
[32m‚úî[39m This task will execute. 

‚Ä∫ I will push an output. [2]
",
  ],
]
`;

exports[`show output from task should output to bottom bar from task with no title persistently: test-e2e-iI52S3WPytorU9EZKPar2AiBrLAZTVut-err 1`] = `Array []`;

exports[`show output from task should output to bottom bar from task with no title persistently: test-e2e-iI52S3WPytorU9EZKPar2AiBrLAZTVut-exit 1`] = `Array []`;

exports[`show output from task should output to bottom bar from task with no title persistently: test-e2e-iI52S3WPytorU9EZKPar2AiBrLAZTVut-out 1`] = `
Array [
  Array [
    "  
",
  ],
  Array [
    "[2K[1A[2K[G 

‚Ä∫ I will push an output. [0] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[G 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2]
",
  ],
]
`;

exports[`show output from task should output to bottom bar from task with no title: test-e2e-JMCRBo4OtLm7JB3XbcYoRcCdQRiKfPdP-err 1`] = `Array []`;

exports[`show output from task should output to bottom bar from task with no title: test-e2e-JMCRBo4OtLm7JB3XbcYoRcCdQRiKfPdP-exit 1`] = `Array []`;

exports[`show output from task should output to bottom bar from task with no title: test-e2e-JMCRBo4OtLm7JB3XbcYoRcCdQRiKfPdP-out 1`] = `
Array [
  Array [
    "  
",
  ],
  Array [
    "[2K[1A[2K[G 

‚Ä∫ I will push an output. [0] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[G 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G  
",
  ],
  Array [
    "[2K[1A[2K[G 
",
  ],
]
`;

exports[`show output from task should output to bottom bar persistently: test-e2e-HhZEM7noGNW4xpgxv4ZtXsPMroPWqrEA-err 1`] = `Array []`;

exports[`show output from task should output to bottom bar persistently: test-e2e-HhZEM7noGNW4xpgxv4ZtXsPMroPWqrEA-exit 1`] = `Array []`;

exports[`show output from task should output to bottom bar persistently: test-e2e-HhZEM7noGNW4xpgxv4ZtXsPMroPWqrEA-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute. 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute. 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2]
",
  ],
]
`;

exports[`show output from task should output to bottom bar: test-e2e-767BkeBTfR1lrS2ANYYH7CLWPATxqyat-err 1`] = `Array []`;

exports[`show output from task should output to bottom bar: test-e2e-767BkeBTfR1lrS2ANYYH7CLWPATxqyat-exit 1`] = `Array []`;

exports[`show output from task should output to bottom bar: test-e2e-767BkeBTfR1lrS2ANYYH7CLWPATxqyat-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute. 

‚Ä∫ I will push an output. [0]
‚Ä∫ I will push an output. [1]
‚Ä∫ I will push an output. [2] 
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[32m‚úî[39m This task will execute. 
",
  ],
]
`;

exports[`show output from task should output to from an observable: test-e2e-SM8IHVdptzrFs7Qk2bseYbdCwtTf03QT-err 1`] = `Array []`;

exports[`show output from task should output to from an observable: test-e2e-SM8IHVdptzrFs7Qk2bseYbdCwtTf03QT-exit 1`] = `Array []`;

exports[`show output from task should output to from an observable: test-e2e-SM8IHVdptzrFs7Qk2bseYbdCwtTf03QT-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m Observable test.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m Observable test.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m Observable test.
  ‚Ä∫ test  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[33m‚ùØ[39m Observable test.
  ‚Ä∫ changed  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[32m‚úî[39m Observable test.  
",
  ],
  Array [
    "[2K[1A[2K[G[32m‚úî[39m Observable test. 
",
  ],
]
`;

exports[`show output from task should yield example output: test-e2e-rMG224TBrLk3ocYtKidc1D4AyZtEHm11-err 1`] = `Array []`;

exports[`show output from task should yield example output: test-e2e-rMG224TBrLk3ocYtKidc1D4AyZtEHm11-exit 1`] = `Array []`;

exports[`show output from task should yield example output: test-e2e-rMG224TBrLk3ocYtKidc1D4AyZtEHm11-out 1`] = `
Array [
  Array [
    "[2m‚óº[22m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
  ‚Ä∫ I will push an output. [0]  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
  ‚Ä∫ I will push an output. [1]  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[33m‚ùØ[39m This task will execute.
  ‚Ä∫ I will push an output. [2]  
",
  ],
  Array [
    "[2K[1A[2K[1A[2K[G[32m‚úî[39m This task will execute.  
",
  ],
  Array [
    "[2K[1A[2K[G[32m‚úî[39m This task will execute. 
",
  ],
]
`;
